app/components/MyComponent.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState, useEffect, useRef } from "react";
+import {
+  Text,
+  View,
+  TouchableOpacity,
+  FlatList,
+  TextInput,
+} from "react-native";
+
+const MyComponent = ({ data }) => {
+  const [selectedItems, setSelectedItems] = useState([]);
```

Comment:
Define type of `selectedItems` state

```suggestion
  const [selectedItems, setSelectedItems] = useState<DataType[]>([]);
```